File: src/main/java/tomeko/screenshotmessageenhancer/config/ScreenshotMessageEnhancerConfig.java
package tomeko.screenshotmessageenhancer.config;

import dev.isxander.yacl3.api.*;
import dev.isxander.yacl3.api.controller.TickBoxControllerBuilder;
import dev.isxander.yacl3.config.v2.api.ConfigClassHandler;
import dev.isxander.yacl3.config.v2.api.SerialEntry;
import dev.isxander.yacl3.config.v2.api.serializer.GsonConfigSerializerBuilder;
import dev.isxander.yacl3.platform.YACLPlatform;
import net.minecraft.client.gui.screens.Screen;
import net.minecraft.network.chat.Component;
import tomeko.screenshotmessageenhancer.utils.Constants;

public class ScreenshotMessageEnhancerConfig {
    public static final ConfigClassHandler<ScreenshotMessageEnhancerConfig> CONFIG = ConfigClassHandler.createBuilder(ScreenshotMessageEnhancerConfig.class)
            .serializer(config -> GsonConfigSerializerBuilder.create(config)
                    .setPath(YACLPlatform.getConfigDir().resolve(Constants.MOD_ID + ".json"))
                    .build())
            .build();

    @SerialEntry
    public static boolean modifyScreenshotMessageEnabled = true;
    @SerialEntry
    public static boolean modifyScreenshotMessageAddName = false;
    @SerialEntry
    public static boolean modifyScreenshotMessageAddCopy = true;
    @SerialEntry
    public static boolean modifyScreenshotMessageAddOpen = true;
    @SerialEntry
    public static boolean modifyScreenshotMessageAddOpenFolder = true;
    @SerialEntry
    public static boolean modifyScreenshotMessageAddDelete = true;

    public static Screen configScreen(Screen parent) {
        return YetAnotherConfigLib.create(CONFIG, ((defaults, config, builder) -> builder
                .title(Component.literal(Constants.MOD_NAME))

                .category(ConfigCategory.createBuilder()
                        .name(Component.literal("Screenshot Message Enhancer Config"))

                        .option(Option.<Boolean>createBuilder()
                                .name(Component.literal("Enabled"))
                                .binding(defaults.modifyScreenshotMessageEnabled, () -> config.modifyScreenshotMessageEnabled, newVal -> config.modifyScreenshotMessageEnabled = newVal)
                                .controller(TickBoxControllerBuilder::create)
                                .build())
                        .option(Option.<Boolean>createBuilder()
                                .name(Component.literal("Add Screenshot Name"))
                                .binding(defaults.modifyScreenshotMessageAddName, () -> config.modifyScreenshotMessageAddName, newVal -> config.modifyScreenshotMessageAddName = newVal)
                                .controller(TickBoxControllerBuilder::create)
                                .build())
                        .option(Option.<Boolean>createBuilder()
                                .name(Component.literal("Add [COPY] Button"))
                                .binding(defaults.modifyScreenshotMessageAddCopy, () -> config.modifyScreenshotMessageAddCopy, newVal -> config.modifyScreenshotMessageAddCopy = newVal)
                                .controller(TickBoxControllerBuilder::create)
                                .build())
                        .option(Option.<Boolean>createBuilder()
                                .name(Component.literal("Add [OPEN] Button"))
                                .binding(defaults.modifyScreenshotMessageAddOpen, () -> config.modifyScreenshotMessageAddOpen, newVal -> config.modifyScreenshotMessageAddOpen = newVal)
                                .controller(TickBoxControllerBuilder::create)
                                .build())
                        .option(Option.<Boolean>createBuilder()
                                .name(Component.literal("Add [OPEN FOLDER] Button"))
                                .binding(defaults.modifyScreenshotMessageAddOpenFolder, () -> config.modifyScreenshotMessageAddOpenFolder, newVal -> config.modifyScreenshotMessageAddOpenFolder = newVal)
                                .controller(TickBoxControllerBuilder::create)
                                .build())
                        .option(Option.<Boolean>createBuilder()
                                .name(Component.literal("Add [DELETE] Button"))
                                .binding(defaults.modifyScreenshotMessageAddDelete, () -> config.modifyScreenshotMessageAddDelete, newVal -> config.modifyScreenshotMessageAddDelete = newVal)
                                .controller(TickBoxControllerBuilder::create)
                                .build())
                        .build())

        )).generateScreen(parent);
    }

    public static void register() {
        ScreenshotMessageEnhancerConfig.CONFIG.load();
    }
}

File: src/main/kotlin/tomeko/screenshotmessageenhancer/config/ScreenshotMessageEnhancerConfig.kt
package tomeko.screenshotmessageenhancer.config

//? if forge {
/*import cc.polyfrost.oneconfig.config.Config
import cc.polyfrost.oneconfig.config.annotations.Exclude
import cc.polyfrost.oneconfig.config.annotations.Info
import cc.polyfrost.oneconfig.config.annotations.Switch
import cc.polyfrost.oneconfig.config.data.InfoType
import cc.polyfrost.oneconfig.config.data.Mod
import cc.polyfrost.oneconfig.config.data.ModType
*///?} else {
import org.polyfrost.oneconfig.api.config.v1.Config
import org.polyfrost.oneconfig.api.config.v1.annotations.Info
import org.polyfrost.oneconfig.api.config.v1.annotations.MultiSelectDropdown
import org.polyfrost.oneconfig.api.config.v1.annotations.Switch
//?}
import tomeko.screenshotmessageenhancer.utils.Constants

object ScreenshotMessageEnhancerConfig : Config(
    //? if forge {
    /*Mod(
        Constants.MOD_NAME,
        ModType.UTIL_QOL,
        Constants.MOD_ICON
    ),
    "${Constants.MOD_ID}.json"
    *///?} else {
    "${Constants.MOD_ID}.json",
    Constants.MOD_ICON,
    Constants.MOD_NAME,
    Category.QOL
    //?}
) {
    fun register() {
        //? if forge {
        //initialize()
        //?} else {
        preload()
        //?}
    }

    //? if forge {
    //@Exclude
    //?}
    private const val CATEGORY_GENERAL = "General"

    //? if forge {
    //@Exclude
    //?}
    private const val SUBCATEGORY_NAME = "Name"

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Show Name",
        description = "Show screenshot name in the screenshot message",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_NAME
    )
    var showName = false

    //? if forge {
    //@Exclude
    //?}
    private const val SUBCATEGORY_BUTTONS = "Buttons"

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Show Copy Button",
        description = "Show copy button in the screenshot message",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_BUTTONS
    )
    var showCopyButton = true

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Show Open Button",
        description = "Show open button in the screenshot message",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_BUTTONS
    )
    var showOpenButton = true

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Show Open Folder Button",
        description = "Show open folder button in the screenshot message",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_BUTTONS
    )
    var showOpenFolderButton = true

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Show Delete Button",
        description = "Show delete button in the screenshot message",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_BUTTONS
    )
    var showDeleteButton = true

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Show Upload Button",
        description = "Show upload button in the screenshot message",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_BUTTONS
    )
    var showUploadButton = true

    //? if forge {
    //@Exclude
    //?}
    private const val SUBCATEGORY_MISC = "Misc"

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Automatically Copy Screenshot",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_MISC
    )
    var autoCopyScreenshot = false

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Compress Screenshots",
        description = "Compress Screenshots without any quality loss",
        category = CATEGORY_GENERAL,
        subcategory = SUBCATEGORY_MISC
    )
    var compressScreenshots = true


    //? if forge {
    //@Exclude
    //?}
    private const val CATEGORY_DEBUG = "Debug"

    @Info(
        //? if forge {
        //text =
            //?} else {
            title =
            //?}
            "Probably should stay disabled",
        //? if forge {
        //type = InfoType.INFO,
        //?}
        category = CATEGORY_DEBUG
    )
    var debugModeInfo: Nothing? = null

    @Switch(
        //? if forge {
        //name =
            //?} else {
            title =
            //?}
            "Debug Mode",
        category = CATEGORY_DEBUG
    )
    var debugModeEnabled: Boolean = false
}